timely/src/progress/timestamp.rs

```diff
@@ -9,10 +9,13 @@ use crate::communication::Data;
 use crate::order::PartialOrder;
 
 /// A composite trait for types that serve as timestamps in timely dataflow.
+///
+/// By implementing this trait, you promise that the type's [PartialOrder] implementation
+/// is compatible with [Ord], such that if `a.less_equal(b)` then `a <= b`.
```

Comment:
Nit, but would it be better to say that by implementing the `PartialOrder` trait you have this requirement? It's a bit awkward because `PartialOrder` does not require `Eq`, or `Ord`, but various uses of it do assume the implied relationship (though not with `PartialOrd`). I'm thinking a beefed up `PartialOrder` doc makes more sense than this comment here (and you have that as part of the PR, but we might want to add even more to it, e.g. wrt `Ord`).